src/internet_identity/src/main.rs

```diff
@@ -813,6 +813,31 @@ fn security_headers() -> Vec<HeaderField> {
     vec![
         ("X-Frame-Options".to_string(), "DENY".to_string()),
         ("X-Content-Type-Options".to_string(), "nosniff".to_string()),
+
+        // Note: trusted types cannot be used due to firefox not supporting SRI.
+        // See https://bugzilla.mozilla.org/show_bug.cgi?id=1409200.
+        //
+        // script-src 'unsafe-eval' is required because agent-js uses a WebAssembly module for bls signatures.
+        // There is currently no other way to allow execution of WebAssembly modules with CSP.
+        // See https://github.com/WebAssembly/content-security-policy/blob/main/proposals/CSP.md and
+        // https://dfinity.atlassian.net/browse/FOLLOW-383.
+        //
+        // script-src 'unsafe-inline' https: are only there for backwards compatibility and ignored
+        // by modern browsers.
+        (
+            "Content-Security-Policy".to_string(),
+            "default-src 'none';\
+             connect-src 'self';\
+             img-src 'self' data:;\
+             script-src 'sha256-syYd+YuWeLD80uCtKwbaGoGom63a0pZE5KqgtA7W1d8=' 'unsafe-inline' 'unsafe-eval' 'strict-dynamic' https:;\
```

Comment:
How did you come up with the hash? Doesn't it depend on the content of the compiled javascript, i.e. don't we have to update this on every change?

---

Actually, is this the hash of the inline script above?

``` js
      const scripts = ["index.js"];
       scripts.forEach(function (scriptUrl) {
         let s = document.createElement("script");
         s.async = false;
         s.src = scriptUrl;
         document.head.appendChild(s);
       });
```

and if so, doesn't this make the whole CSP header moot? I thought we wanted to bake in the hashes of the loaded javascript files. Sorry, noob here 😅

Reply:
Yes, it is the hash of this inline script (so we do not have to update for every change). This does **not** make the CSP moot for the following reasons:
* The CSP still disallows all scripts that are 'parser-inserted'. See here: https://www.w3.org/TR/CSP3/#strict-dynamic-usage
* In our case, we have additional guarantees as those scripts are certified by IC.
* The CSP also has directives unrelated to scripts which improve security as well.

This workaround is recommended [here](https://csp.withgoogle.com/docs/faq.html#static-content) for static pages that cannot use nonces (such as II). But I agree, it would be nicer if we could just rely on SRI. But unfortunately Firefox does not (yet) support this.